### DISCUSSION TITLE
Multiple Domain / Site Integration Feature Request

### DISCUSSION THREAD

**Discussion by l-arnold** [joomla/joomla-cms#47178] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
Potential Multi-Domain/Site setup in Core Joomla

Prior Closed Issue had no resolution and, from an outside view, quite a bit of infighting.  The requestor put up the same request I am looking for and he had a very good set of evaluations of the problem.  Certainly did not deserve to be closed.  I see other issues still open from 2017 still now.
https://github.com/joomla/joomla-cms/issues/22254

### Describe the solution you'd like

Just as ACL can be extended,  Domain/Site basis should be extendable.

### Additional context

Complicated to do this now.  Nothing "easy" in the back end, and logical processes of different extensions are complicated and not well documented.
VirtualDomains has been dropped from Joomla Extensions for some reason though it has functionality still (but limited documentation)
Joomla documentation on how to do this out of the box is very code dependent and not built on the simple admin context of most of Joomla Admin.

MightySites, Multi-sites and Seybold seem to have ways.  Seybold system seems similar to VirtualDomains.

It would seem that with the soon to come Joomla4 this would be an excellent opportunity to integrate such a system into the standard build.

Otherwise it can be very wasteful/expensive/complicated to run multiple related Joomla sites.

Feature Request

**Comment by HLeithner** [comment]
Hi @l-arnold 

at this point in time there is nobody (I'm aware of ) that works on such a feature. 
I'm also really interested in a multi domain / multi site feature too.

But sadly to say this will not come to Joomla 4.0.

**Comment by ElephantInTheRoomFirst** [comment]
The feature is interesting. I am not sure that how many percentage of Joomla users need the feature?
You can achieve some functionality using fabrik.

**Comment by ghost** [comment]
1)
The poster of this thread believes the concept I have proposed has a meaning.

That is true.

I have worked with multisite since many years and have understood its inherent problems. Not only that, I  analysed multisite concepts of other CMS/php scripts like SilverStripe, etc. I know this area very good. As a result of this experience, I  proposed a concept on 19 Sep. 2018. Considering the central idea of this concept, there is none better than the one I had proposed.

https://github.com/joomla/joomla-cms/issues/22254

I was surprised to see reactions by babker and brianteeman, who vehemently opposed it. They did not understand it's terrific potential and ease of use, although they expressed to understand it. Unless a person works with a multisite concept and understand it's inherent problems, it is not possible to develop an excellent concept on a multidomain or multisubdomain approach.

**Thus, I would like to ask the developers to re-open my thread and make a discussion there. Then, we could debate on this topic in depth.**

I would like to explain why this concept could be implemented in a case, where a user is not an administrator, have less experience or does not have root access  in a shared hosting environment. They do not need special rights, which both the opposers expressed in there.

In the alternative, I could join the Group, who is or will be responsible to develop this area of multisite.

2)
It is blatantly wrong to declare that there are not many users. Implement this feature and everyone will use it. In business, majority uses multiple domains or subdomains. Either they have different domains with different content or they use subdomains with content in different languages.

Leading CMS like Drupal, Wordpress, SilverStripe, etc. implemented multisite feature before years. There are many changes in the core, which allowed implementation of an extension.

3)
It is not possible to achieve a true multisite feature by fabrik. Just by changing content language on the fly is not a solution. 

4)
A true multisite solution would be to create separated content based on the same php installation. Here there are three fundamental approaches:

a. Have different tables in different databases. The scripts then connects to different DB. This has an inherent lacuna.

b. Have tables with different prefixes. Even here, there are limitations.

c. Use a hybrid approach in both above.

All three solutions above with get into its own limitations. By using the SQL views, these limitations are drastically reduced. The biggest advantage with use of SQL views is to share data across websites that have common parameters, like tables for Regions, countries, states, cities, zip, etc. Thus, a concept is required that is built on these hybrid solution. By doing so, the inherent limitation in the Joomla core is reduced too.

**Comment by brianteeman** [comment]
The responses you received three years ago are just as valid today

**Comment by ghost** [comment]
Then let be as it should be and as it is!

**Comment by ghost** [comment]
Brian, so if the responses are the same, why not close this thread as well?

**Comment by brianteeman** [comment]
Because other people may have more practical and achievable suggestions than yours

**Comment by l-arnold** [comment]
I've seen official comments floating by.  Here are a few of them:
[https://docs.joomla.org/Multi-site_Working_Group]  (https://docs.joomla.org/Multi-site_Working_Group)

Multi Site is also noted here:
[https://www.joomla.org/announcements/release-news/5819-the-day-after-joomla-4-0-planning-the-future.html](url)

Mentioned specifically towards version 4.2:  Main-Feature: Multi-Domain
Implement a multi-Domain/Site functionality into Joomla! we are forming the team at the moment, if you’re interested to help please contact the Multidomain working group.

**Comment by l-arnold** [comment]
I do have Virtual Domains working..  takes some work to be sure to keep it all straight.

**Comment by wijzijnkamels** [comment]
As 4.2 is the current production version, what is the status of Multi-Domain as mentioned in this article (https://www.joomla.org/announcements/release-news/5819-the-day-after-joomla-4-0-planning-the-future.html)?

**Comment by ReLater** [comment]
> As 4.2 is the current production version, what is the status of Multi-Domain as mentioned in this article ...?

> Main-Feature: Multi-Domain Implement a multi-Domain/Site functionality into Joomla! we are forming the team at the moment, if you’re interested to help please contact the [Multidomain working group](https://volunteers.joomla.org/teams/multidomain-working-group).

Instead of posting redundant stuff here participate, start programming, provide testable code for real life environments! Contact the working group members!
